src/xcode/ENA/ENA/Source/Scenes/HealthCertificate/HealthCertificate/HealthCertificateViewModel.swift

```diff
@@ -0,0 +1,243 @@
+			.sink(
+				receiveCompletion: { result in
+					switch result {
+					case .finished:
+						break
+					case .failure(let error):
+						if case CachingHTTPClient.CacheError.dataVerificationError = error {
+							Log.error("Signature verification error.", log: .vaccination, error: error)
+						}
+						Log.error("Could not fetch Vaccination value sets protobuf.", log: .vaccination, error: error)
+					}
+				}, receiveValue: { [weak self] valueSets in
+					self?.valueSets = valueSets
+					self?.setupHealthCertificateKeyValueCellViewModel()
+				}
+			)
+			.store(in: &subscriptions)
+
+	}
+
+	// MARK: - Internal
+
+	enum TableViewSection: Int, CaseIterable {
+		case headline
+		case qrCode
+		case topCorner
+		case details
+		case bottomCorner
+
+		static var numberOfSections: Int {
+			allCases.count
+		}
+
+		static func map(_ section: Int) -> TableViewSection? {
+			guard let section = TableViewSection(rawValue: section) else {
+				Log.error("unknown TableViewSection", log: .vaccination)
+				return nil
+			}
+			return section
+		}
+	}
+
+	let qrCodeCellViewModel: HealthCertificateQRCodeCellViewModel
+
+	@OpenCombine.Published private(set) var gradientType: GradientView.GradientType = .solidGrey
+	@OpenCombine.Published private(set) var healthCertificateKeyValueCellViewModel: [HealthCertificateKeyValueCellViewModel] = []
+
+	var headlineCellViewModel: HealthCertificateSimpleTextCellViewModel {
+		let centerParagraphStyle = NSMutableParagraphStyle()
+		centerParagraphStyle.alignment = .center
+		centerParagraphStyle.lineSpacing = 10.0
+
+		let attributedName = NSAttributedString(
+			string: String(format: AppStrings.HealthCertificate.Details.vaccinationCount, healthCertificate.doseNumber, healthCertificate.totalSeriesOfDoses),
+			attributes: [
+				.font: UIFont.enaFont(for: .headline),
+				.foregroundColor: UIColor.enaColor(for: .textContrast),
+				.paragraphStyle: centerParagraphStyle
+			]
+		)
+
+		let attributedDetails = NSAttributedString(
+			string: AppStrings.HealthCertificate.Details.certificate,
+			attributes: [
+				.font: UIFont.enaFont(for: .body),
+				.foregroundColor: UIColor.enaColor(for: .textContrast),
+				.paragraphStyle: centerParagraphStyle
+			]
+		)
+
+		return HealthCertificateSimpleTextCellViewModel(
+			backgroundColor: .clear,
+			textAlignment: .center,
+			attributedText: [attributedName, attributedDetails]
+				.joined(with: "\n"),
+			topSpace: 18.0,
+			font: .enaFont(for: .headline),
+			accessibilityTraits: .staticText
+		)
+	}
+
+	func numberOfItems(in section: TableViewSection) -> Int {
+		switch section {
+		case .headline:
+			return healthCertificate.vaccinationCertificates.isEmpty ? 0 : 1
+		case .details:
+			return healthCertificateKeyValueCellViewModel.count
+		default:
+			return healthCertificateKeyValueCellViewModel.isEmpty ? 0 : 1
+		}
+	}
+
+	// MARK: - Private
+
+	private enum ValueSetType: String {
+		case vp
```

Comment:
Can we have more self explanatory cases or this was the requirement?

Reply:
Or perhaps a comment on top, what is vp, mp and ma?

Reply:
I use the longer names now here as well, as we do in the certificate.